src/test/java/org/apache/commons/collections4/bloomfilter/LayerManagerTest.java

```diff
@@ -0,0 +1,305 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+public class LayerManagerTest {
+
+    private Shape shape = Shape.fromKM(17, 72);
+
+    private LayerManager.Builder testingBuilder() {
+        return LayerManager.builder().setSupplier(() -> new SimpleBloomFilter(shape));
+    }
+
+    @Test
+    public void testAdvanceOnPopulated() {
+        Predicate<LayerManager> underTest = LayerManager.ExtendCheck.advanceOnPopulated();
+        LayerManager layerManager = testingBuilder().build();
+        assertFalse(underTest.test(layerManager));
+        layerManager.getTarget().merge(TestingHashers.FROM1);
+        assertTrue(underTest.test(layerManager));
+    }
+
+    @Test
+    public void testNeverAdvance() {
+        Predicate<LayerManager> underTest = LayerManager.ExtendCheck.neverAdvance();
+        LayerManager layerManager = testingBuilder().build();
+        assertFalse(underTest.test(layerManager));
+        for (int i = 0; i < 10; i++) {
+            layerManager.getTarget().merge(TestingHashers.randomHasher());
+            assertFalse(underTest.test(layerManager));
+        }
+    }
+
+    private void testAdvanceOnCount(int breakAt) {
+        Predicate<LayerManager> underTest = LayerManager.ExtendCheck.advanceOnCount(breakAt);
+        LayerManager layerManager = testingBuilder().build();
+        for (int i = 0; i < breakAt - 1; i++) {
+            assertFalse(underTest.test(layerManager), "at " + i);
+            layerManager.getTarget().merge(TestingHashers.FROM1);
+        }
+        assertTrue(underTest.test(layerManager));
+    }
+
+    @Test
+    public void testAdvanceOnCount() {
+        testAdvanceOnCount(4);
+        testAdvanceOnCount(10);
+        testAdvanceOnCount(2);
+        testAdvanceOnCount(1);
+        assertThrows(IllegalArgumentException.class, () -> LayerManager.ExtendCheck.advanceOnCount(0));
+        assertThrows(IllegalArgumentException.class, () -> LayerManager.ExtendCheck.advanceOnCount(-1));
+    }
+
+    @Test
+    public void testAdvanceOnCalculatedFull() {
+        Double maxN = shape.estimateMaxN();
+        Predicate<LayerManager> underTest = LayerManager.ExtendCheck.advanceOnSaturation(shape.estimateMaxN());
+        LayerManager layerManager = testingBuilder().build();
+        while (layerManager.getTarget().getShape().estimateN(layerManager.getTarget().cardinality()) < maxN) {
+            assertFalse(underTest.test(layerManager));
+            layerManager.getTarget().merge(TestingHashers.randomHasher());
+        }
+        assertTrue(underTest.test(layerManager));
+    }
+
+    @Test
+    public void testAdvanceOnSaturation() {
```

Comment:
This is exactly the same as `testAdvanceOnCalculatedFull`. I think this should be:
```java
double maxN = shape.estimateMaxN();
Predicate<LayerManager> underTest = LayerManager.ExtendCheck.advanceOnSaturation(maxN);
```